packages/odyssey-react-mui/src/labs/DataComponents/TableSettings.tsx

```diff
@@ -113,15 +108,13 @@ const TableSettings = ({
         >
           <>
             {columns
-              .filter((column) => column.enableHiding !== false)
+              .filter((column) => Boolean(column.enableHiding))
               .map((column) => (
                 <MenuItem
                   key={column.accessorKey}
-                  onClick={changeColumnVisibility(column.id as string)}
+                  onClick={() => changeColumnVisibility(column.id!)}
                 >
-                  <MuiCheckbox
-                    checked={isColumnVisibilityChecked[column.accessorKey!]}
-                  />
+                  <MuiCheckbox checked={visibleColumns.has(column.id!)} />
```

Comment:
Does it think we don't always have a column `id` for some reason?

Reply:
Good catch — it's not guaranteed to have an id, but it's guaranteed to have either an id or an accessorKey, as defined by the `MRT_ColumnDef` type that we're following here. I've modified this to match the type properly.